blockchain/vm/instructions.go

```diff
@@ -482,8 +482,7 @@ func opDifficulty(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext)
 func opRandom(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
 	// evm.BlockNumber.Uint64() is always greater than or equal to 1
 	// since evm will not run on the genesis block
-	prevBlockHash := interpreter.evm.Context.GetHash(interpreter.evm.Context.BlockNumber.Uint64() - 1)
-	v, _ := uint256.FromBig(prevBlockHash.Big())
+	v, _ := uint256.FromBig(interpreter.evm.Context.Random.Big())
```

Comment:
Before randao hardfork, it needs to return blockHash of previous block. What about make a new opRandom function for randao fork? eg. `opRandomRandaofork`

Reply:
The `blockchain/evm.go:NewEVMBlockContext` function assigns the `Context.Random` field to either `header.MixHash` or `header.ParentHash` depending on fork status. It is analogous to how we assign `Context.BaseFee` depending on fork status.

Reply:
Oh.. you're right. I misunderstood it. Thank you!